addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -564,11 +567,12 @@ export function createRelatedModels(modelDefs, modelClasses = {}, indexes = {})
                 if (!indexes[model].includes(key)) {
                     throw new Error(`Unable to get record by '${key}'`);
                 }
-                const result = this.indexedRecords[model][key][val];
-                if (result instanceof Map) {
-                    return Array.from(result.values());
+
+                if (X2MANY_TYPES.has(fields[key].type)) {
+                    return Array.from(this.indexedRecords[model][key][val]?.values() || []);
+                } else {
+                    return this.indexedRecords[model][key][val];
```

Comment:
Do you really need to make this change? I don't see any link to the `orderedRecords`. Can you tell me how it's linked?